File: 5. OOP/06. EncapsulationAndPolymorphism/P01Shapes/Circle.cs
namespace P01Shapes
{
    using System;

    public class Circle : BasicShape
    {
        public Circle(double width) : base(width, width)
        {
        }

        public override double CalculateArea()
        {
            double area = Math.PI * this.Width * this.Width / 4;
            return area;
        }

        public override double CalculatePerimeter()
        {
            double perimeter = Math.PI * 2 * this.Width / 2;
            return perimeter;
        }
    }
}

File: 1. JavaBasics/P01Geometry/src/planeShapes/Circle.java
package planeShapes;

import abstractShapes.*;

public class Circle extends PlaneShape {

    private double radius;

    public Circle(Vertex[] vertices, double radius) {
	super(vertices);
	this.setRadius(radius);
    }

    public double getRadius() {
	return radius;
    }

    public void setRadius(double radius) throws IllegalArgumentException {
	if (radius < 0) {
	    throw new IllegalArgumentException("The radius can't be negative!");
	}

	this.radius = radius;
    }

    @Override
    public double getPerimeter() {
	double perimeter = 2 * Math.PI * this.radius;
	return perimeter;
    }

    @Override
    public double getArea() {
	double area = Math.PI * this.radius * this.radius;
	return area;
    }

}
